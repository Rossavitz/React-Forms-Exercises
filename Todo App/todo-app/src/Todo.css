.Todo{
    display: flex;
    justify-content: center;
    align-items: center;
}

.Todo-task{
    margin: 3px;
}
