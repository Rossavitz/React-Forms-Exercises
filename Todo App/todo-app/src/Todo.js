import React from "react";
import "./Todo.css";

const Todo = ({ task, id, handleRemove }) => {
  const removeTodo = () => {
    handleRemove(id);
  };
  return (
    <div className="Todo">
      <p className="Todo-task">{task}</p>
      <button onClick={removeTodo} className="Todo-btn">
        x
      </button>
    </div>
  );
};

export default Todo;
