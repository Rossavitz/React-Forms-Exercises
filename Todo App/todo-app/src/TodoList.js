import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
  const INITIAL_STATE = [];

  const [todos, setTodos] = useState(INITIAL_STATE);
  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
  };

  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  const todo = todos.map(({ id, task, handleRemove }) => (
    <Todo key={id} id={id} task={task} handleRemove={removeTodo} />
  ));

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      {todo}
    </div>
  );
};

export default TodoList;
