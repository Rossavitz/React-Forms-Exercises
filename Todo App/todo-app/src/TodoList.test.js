import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";
import Todo from "./Todo";

// smoke test
it("renders without crashing", function () {
  render(<TodoList />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a todo", function () {
  render(<TodoList />);
  const input = screen.queryByLabelText("Add Todo:");
  const btn = screen.queryByText("Add Todo");
  expect(screen.queryByText("walk dogs")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "walk dogs" } });
  fireEvent.click(btn);
  expect(screen.getByText("walk dogs")).toBeInTheDocument();
});

it("should delete a todo", function () {
  render(<TodoList />);
  const input = screen.queryByLabelText("Add Todo:");
  const btn = screen.queryByText("Add Todo");
  expect(screen.queryByText("walk dogs")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "walk dogs" } });
  fireEvent.click(btn);
  expect(screen.getByText("walk dogs")).toBeInTheDocument();
  const removeBtn = screen.queryByText("x");
  fireEvent.click(removeBtn);
  expect(screen.queryByText("walk dogs")).not.toBeInTheDocument();
});
