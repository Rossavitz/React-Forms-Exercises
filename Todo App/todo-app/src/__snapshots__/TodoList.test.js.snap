// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="Add-Todo"
      >
        Add Todo: 
      </label>
      <input
        id="Add-Todo"
        name="task"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
